mlos_bench/mlos_bench/services/remote/azure/azure_services.py

```diff
@@ -291,12 +309,16 @@ def _check_vm_operation_status(self, params: dict) -> Tuple[Status, dict]:
         if url is None:
             return Status.PENDING, {}
 
+        session = self._get_session(params)
         try:
-            response = requests.get(url, headers=self._get_headers(), timeout=self._request_timeout)
+            response = session.get(url, headers=self._get_headers(), timeout=self._request_timeout)
```

Comment:
we can probably move `headers=self._get_headers()` into the Session, too?

Reply:
Good point, moved the headers into Session too!